src/AWS.Deploy.Common/Recommendation.cs

```diff
@@ -53,8 +53,37 @@ public Recommendation(RecipeDefinition recipe, ProjectDefinition projectDefiniti
             }
         }
 
+        public List<Category> GetConfigurableOptionSettingCategories()
+        {
+            var categories = Recipe.Categories;
+
+            // If any top level settings has a category of General make sure the General category is added to the list.
+            if(!categories.Any(x => string.Equals(x.Id, Category.General.Id)) &&
+                Recipe.OptionSettings.Any(x => string.IsNullOrEmpty(x.Category) || string.Equals(x.Category, Category.General.Id)))
+            {
+                categories.Insert(0, Category.General);
+            }
+
+            // Add the build settings category if it is not already in the list of categories.
+            if(!categories.Any(x => string.Equals(x.Id, Category.DeploymentBundle.Id)))
+            {
+                categories.Add(Category.DeploymentBundle);
+            }
+
+            return categories;
+        }
+
         public IEnumerable<OptionSettingItem> GetConfigurableOptionSettingItems()
         {
+            // Make sure at least all top level settings have a category of General
+            foreach(var setting in Recipe.OptionSettings)
+            {
+                if(string.IsNullOrEmpty(setting.Id))

```

Comment:
didn't you mean to check if a setting does not have a category? 
Also the comment on line 78 in confusing. Maybe just say that you're adding the general category for settings that don't have categories

Reply:
Fixed